recipe-app/ViewModel/RecipeListViewModel.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  RecipeListViewModel.swift
+//  recipe-app
+//
+//  Created by Arthur Hermann on 27/11/2024.
+//
+
+import UIKit
+
+final class RecipeListViewModel {
+    private let apiKey = "5b91aa819e6d4bc8848f4c972103e6dc"
+    var recipes = [Recipe]()
+    
+    private func makeURL(query: String) -> URL? {
+        var components = URLComponents(string: "https://api.spoonacular.com/recipes/complexSearch")
+        components?.queryItems = [
+            URLQueryItem(name: "apiKey", value: apiKey),
+            URLQueryItem(name: "query", value: query),
+            URLQueryItem(name: "number", value: "20")
+        ]
+        return components?.url
+    }
+    //parses JSON files
+    private func parse(json: Data) throws -> [Recipe] {
+        do {
+            let decoder = JSONDecoder()
+            let jsonRecipes = try decoder.decode(Recipes.self, from: json)
+            recipes = jsonRecipes.results
+            
+        }
+        catch  {
+            RecipeError.parsingError
+        }
+        return recipes
+    }
+    
+    private enum RecipeError: Error {
+        case invalidURL, networkError, unexpectedResponse, noData, parsingError
```

Comment:
Let's move our enums to another file. And please use below format to increase readability.

```suggestion
        case invalidURL
        case networkError
        case unexpectedResponse
        case noData
        case parsingError
```